### DISCUSSION TITLE
Multiple heating circuits / more than 1 central boiler

### DISCUSSION THREAD

**Discussion by shaftspanner** [jmcollin78/versatile_thermostat#712] [Q&A]
Hi all,

I have 2 heating circuits (upstairs and downstairs), each controlled by a simple on/off zigbee switch.  Either or both can trigger my boiler to fire resulting in hot water flowing to the relevant heating circuit.

I'm trying to set up a number of TRVs within each heating circuit, but when I come to setup the central boiler, VTherm only allows me to setup one boiler.  Is it possible to independently trigger each heating circuit within VTherm?

**Comment by jmcollin78** [comment]
Hello @shaftspanner ,

There is only one central boiler possible in VTherm. But I guess you have only one. 
You can control the switchs with 2 VTherms `over_switch` over each switch. If you have a thermometer on each heating zone (upstairs and downstairs), VTherm will open / close the switch until the target temperature is reach.

See https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/over-switch.md

Then you can configure a central boiler which will be activated if one of the switch is on. Maybe you already have this mecanism.

It seems possible for me. To avoid to often switchs you can  configure a large cycle (15 min) and and minimal activation delay to 5 min.
See here: https://github.com/jmcollin78/versatile_thermostat/blob/main/documentation/en/feature-advanced.md#minimum-activation-delay

**Reply by shaftspanner** [reply]
Hey @jmcollin78 thanks for your reply and more importantly, thanks for a really awesome integration.  I've been playing around with different ways of controlling my heating for a while now and when I came across versatile thermostat it's capabilities really blew me away - I'm really hoping I can get it to work for my use case as trying to manually implement everything was turning into a nightmare.

I don't know about other parts of the world, but I think my use case would be pretty common in the UK where all but the smallest houses often have two heating circuits - one controlling upstairs radiatiors, the other controlling downstairs radiators.  Traditionally, each heating circuit has a thermostat that demands heat from a single boiler and a valve determines which circuit gets the heat based on demand.  For the sake of VTherm, I guess this would be represented by 2 (virtual) central boilers even though I only have one physical device.

In my case, I've already replaced the circuit thermostat with a switch, and I have thermostats in each room, so I could trigger heat to a circuit based on a single or average temperature as you suggest above.  However that still sends heat to multiple radiators at a time - to the entire floor.  

What I'm trying to do is provide heat to a single room (1 or more radiators within a single circuit) based on the temperature of that room.   I have:

- Each heating circuit controlled by a switch which enables the demand of heat to that circuit
- Sonoff TRVs that can be fitted to radiators to provide a climate entity in each room
- Temperature sensors in each room
 
If I had a single heating circuit, I think VTherm would work very well using `over_climate` with the TRVs and a central boiler.  I guess what I'm looking for is the ability to use 2 central boilers or to have two separate copies of VTherm running on the same HA instance, though I don't think either of these is possible at the moment.

Can you see any other ways of achieving this?

**Reply by jmcollin78** [reply]
> VTherm running on the same HA instance, though I don't think either of these is possible at the moment.

no, this is not possible. You need two HA instance if you want two Central Boilier in VTherm.

**Reply by shaftspanner** [reply]
OK, many thanks for your response. 

I think I'm going to be spinning up a second instance of HA on my proxmox server so I can keep working on this.

But any alternative suggestions would be very much appreciated!
